### PR TITLE
Common find related stuff

### PR DESCRIPTION
**Pull request by sebv**
implicitWaitForCondition + added locator check.

### CODE REVIEW

**Comment by sebv** [comment]
ping @Jonahss @moizjv @imurchie @jlipps

File: lib/commands/find.js [hunk 1]
```diff
@@ -3,25 +3,29 @@ let commands = {}, helpers = {}, extensions = {};
...
-// stategy: locator strategy
+//helpers.findElOrEls = async function (strategy, selector, mult, context) {}
+// strategy: locator strategy
 // selector: the actual selector for finding an element
 // mult: multiple elements or just one?
 // context: finding an element from the root context? or starting from another element
-//helpers.findElOrEls = async function (strategy, selector, mult, context) {}
 
 commands.findElement = async function (strategy, selector) {
+  this.validateLocatorStrategy(strategy, !!this.context);
```
Comment (jlipps) [inline lib/commands/find.js]: what is `this.context`?

File: lib/commands/timeout.js [hunk 2]
```diff
@@ -13,6 +14,20 @@ commands.timeouts = async function (name, duration) {
   }
 };
 
+commands.implicitWaitForCondition = async function (condFn) {
+  log.debug("Waiting up to " + this.implicitWaitMs + "ms for condition");
```
Comment (jlipps) [inline lib/commands/timeout.js]: interpolated string

File: lib/commands/timeout.js [hunk 3]
```diff
@@ -13,6 +14,20 @@ commands.timeouts = async function (name, duration) {
...
+  let wrappedCondFn = async function (...args) {
+    // reset command timeout
+    // TODO: can we do better than that?
+    if (this.noCommandTimer) {
+      clearTimeout(this.noCommandTimer);
+    }
+    return await condFn(...args);
+  }.bind(this);
```
Comment (jlipps) [inline lib/commands/timeout.js]: why not an arrow function instead of bind?

File: lib/commands/timeout.js [hunk 4]
```diff
@@ -13,6 +14,20 @@ commands.timeouts = async function (name, duration) {
...
+  return await waitForCondition(wrappedCondFn, {
+    waitMs: this.implicitWaitMs, intervalMs: 500 , logger: log
```
Comment (jlipps) [inline lib/commands/timeout.js]: weird space in front of `,`

File: lib/commands/timeout.js [hunk 5]
```diff
@@ -13,6 +14,20 @@ commands.timeouts = async function (name, duration) {
...
+  log.debug("Waiting up to " + this.implicitWaitMs + "ms for condition");
+  let wrappedCondFn = async function (...args) {
+    // reset command timeout
+    // TODO: can we do better than that?
+    if (this.noCommandTimer) {
+      clearTimeout(this.noCommandTimer);
+    }
+    return await condFn(...args);
+  }.bind(this);
+  return await waitForCondition(wrappedCondFn, {
```
Comment (jlipps) [inline lib/commands/timeout.js]: no need to `return await`?

File: lib/commands/timeout.js (hunk 3 above)
Comment (sebv) [inline lib/commands/timeout.js]: jshint doesn't like it.

File: lib/commands/timeout.js (hunk 5 above)
Comment (sebv) [inline lib/commands/timeout.js]: It's better, less chance of regenerator bugs, all the return promises are the same.

File: lib/commands/timeout.js (hunk 3 above)
Comment (sebv) [inline lib/commands/timeout.js]: added TODO

File: lib/commands/find.js (hunk 1 above)
Comment (sebv) [inline lib/commands/find.js]: removed it, overriding validateLocatorStrategy in ios driver, still it feels strange, the webcontext code could be more consistent.

**Comment by sebv** [comment]
fixed stuff.

**Pull request merged**